server/resources/resourcesController.js

```diff
@@ -63,6 +64,27 @@ router
 
 router
 	.route("/:id")
+	.get(
+		asyncHandler(async (req, res) => {
+			const {
+				rows: [resource],
+			} = await db.query(
+				singleLine`
+			SELECT r.*, t.name AS topic_name, us.name AS source_name, up.name AS publisher_name
+			FROM resources AS r
+			INNER JOIN users AS us ON r.source = us.id
+			LEFT JOIN users AS up ON r.publisher = up.id
+			LEFT JOIN topics AS t ON r.topic = t.id
+			WHERE r.id = $1;
+		`,
+				[req.params.id]
+			);
```

Comment:
❓Could we include this in the resources repository file if we are to follow the pattern?

Reply:
✅ Yes, this is definitely in the wrong place, repository access should **not** be in the controller. There's a service/business logic layer missing too, although in this case it wouldn't do a great deal.